scripts/feature_tests/upgrade/combined_tests/1cp_1w_bootDiskImageANDK8sControllers_clusterLevel_upgrade.sh

```diff
@@ -0,0 +1,190 @@
+#!/bin/bash
+
+set -x
+
+IRONIC_IMAGE_TAG=master
+
+METAL3_DEV_ENV_DIR="$(dirname "$(readlink -f "${0}")")/../../../../"
+# shellcheck disable=SC1091
+# shellcheck disable=SC1090
+source "${METAL3_DEV_ENV_DIR}/scripts/feature_tests/upgrade/upgrade_common.sh"
+
+echo '' >~/.ssh/known_hosts
+
+start_logging "${1}"
+# Provision original nodes
+set_number_of_master_node_replicas 1
+set_number_of_worker_node_replicas 1
+
+provision_controlplane_node
+
+controlplane_is_provisioned
+controlplane_has_correct_replicas 1
+
+# apply CNI
+apply_cni
+
+provision_worker_node
+worker_has_correct_replicas 1
+
+# ----------- upgrade controlplane components ---------------
+cleanup_clusterctl_configuration
+
+buildClusterctl
+
+# Install initial version folder structure
+pushd /tmp/cluster-api-clone || exit
+cmd/clusterctl/hack/local-overrides.py
+popd || exit
+
+create_clusterctl_configuration
+
+# Create a new version
```

Comment:
`controllers` is missing? I am wondering, if we should have a comment either for all called functions or not have them at all?

Reply:
removed